plugins/rest/src/generate_token_command.cpp

```diff
@@ -0,0 +1,42 @@
+#include "rest/generate_token_command.hpp"
+
+#include "rest/authenticator.hpp"
+
+#include <vast/system/spawn_or_connect_to_node.hpp>
+
+#include <caf/scoped_actor.hpp>
+
+namespace vast::plugins::rest {
+
+auto generate_token_command(const vast::invocation& inv,
+                            caf::actor_system& system) -> caf::message {
+  auto self = caf::scoped_actor{system};
+  auto node_opt = vast::system::spawn_or_connect_to_node(
+    self, inv.options, content(system.config()));
+  if (auto* err = std::get_if<caf::error>(&node_opt))
+    return caf::make_message(std::move(*err));
+  const auto& node
+    = std::holds_alternative<system::node_actor>(node_opt)
+        ? std::get<system::node_actor>(node_opt)
+        : std::get<scope_linked<system::node_actor>>(node_opt).get();
+  // The typed `get_node_components()` only works for actors whose type id is
+  // defined in the main vast namespace, so we have to work around manually.
+  // TODO: Introduce a `component_plugin::component_name()` function so we
+  // can name the component "authenticator" instead of "rest".
```

Comment:
That name is also used for the top-level entry in `vast status`, so the question becomes whether users find `vast status rest` or `vast status authenticator` easier.

Reply:
Ideally we'd have a separate component called `rest` that displays the actually interesting state (ie. listen address and mode), but I'm not sure that's implementable with the current plugin framework.